xrpl/utils/txn_parser/utils/nodes.py

```diff
@@ -0,0 +1,92 @@
+"""Helper functions to normalize an affected node."""
+
+from typing import List, Optional, Union, cast
+
+from typing_extensions import Literal, TypedDict
+
+from xrpl.models import TransactionMetadata
+from xrpl.models.transactions.metadata import (
+    CreatedNode,
+    DeletedNode,
+    Fields,
+    ModifiedNode,
+)
+
+
+class OptionalFieldNames(TypedDict, total=False):
+    """The optional fields of `NormalizedNode`."""
+
+    """
+    The fields are separated from `NormalizedNode` to make it optional,
+    while keeping `NodeType`, `LedgerEntryType` and `LedgerIndex` required.
+    """
+
+    NewFields: Optional[Fields]
+    FinalFields: Optional[Fields]
+    PreviousFields: Optional[Fields]
+    PreviousTxnID: Optional[str]
+    PreviousTxnLgrSeq: Optional[int]
+
+
+class NormalizedNode(OptionalFieldNames):
+    """A model representing an affected node in a standard format."""
+
+    NodeType: Literal["CreatedNode", "ModifiedNode", "DeletedNode"]
+    LedgerEntryType: str
+    LedgerIndex: str
+
+
+def _normalize_node(
+    affected_node: Union[CreatedNode, ModifiedNode, DeletedNode]
+) -> NormalizedNode:
+    node_keys = affected_node.keys()
+    assert len(node_keys) == 1
+    diff_type = cast(
+        Literal["CreatedNode", "ModifiedNode", "DeletedNode"],
+        list(node_keys)[0],
+    )
+    created_node = None
+    modified_node = None
+    if diff_type == "CreatedNode":
+        created_node = cast(CreatedNode, affected_node)["CreatedNode"]
+    elif diff_type == "ModifiedNode":
+        modified_node = cast(ModifiedNode, affected_node)["ModifiedNode"]
+    else:
+        deleted_node = cast(DeletedNode, affected_node)["DeletedNode"]
+    node = (
+        created_node
+        if created_node is not None
+        else modified_node
+        if modified_node is not None
+        else deleted_node
```

Comment:
I think this could be cleaner using if/else
```suggestion
if (created_node is not None):
    node = created_node
elif (modified_node is not None):
    node = modified_node
else: 
    node = deleted_node
```

Reply:
Actually, it's probably easier to just set `node` directly instead of setting `created_node`/`modified_node`/`deleted_node` as intermediary values.

Reply:
I would have done so, but unfortunately mypy raises: 'Incompatible types in assignment (expression has type "ModifiedNodeFields", variable has type "CreatedNodeFields")'

Reply:
Your suggestion is also not valid. Same error.

Reply:
Ah sorry about that. Since node's type isn't inferrable from it's first declaration we have to specify the type explicitly. I tried locally and this version works:
```
    if diff_type == "CreatedNode":
        node: Union[CreatedNodeFields, ModifiedNodeFields, DeletedNodeFields] = cast(
            CreatedNode, affected_node
        )["CreatedNode"]
    elif diff_type == "ModifiedNode":
        node = cast(ModifiedNode, affected_node)["ModifiedNode"]
    else:
        node = cast(DeletedNode, affected_node)["DeletedNode"]
```

(Note that this replaces the `diff_type` if statements AND the `node = ...` statement)

Reply:
I don't have a preference either way. I think the TypedDicts make sense, and that using classes would also make sense. So I'm leaning towards what's easier.